lbry/lbry/stream/descriptor.py

```diff
@@ -46,6 +58,19 @@ def file_reader(file_path: str):
             offset += bytes_to_read
 
 
+def sanitize_file_name(dirty_name: str):
+    file_name, ext = os.path.splitext(dirty_name)
+    file_name = re.sub(RES, '', file_name)
+    ext = re.sub(RES, '', ext)
+
+    if not file_name:
+        log.warning('Unable to suggest a file name for %s', dirty_name)
```

Comment:
In what situation would this happen? Should it maybe return the `dirty_name` in this case? Should the function throw an error?

Reply:
When after removing all illegal characters, the file name is still empty. We may return `dirty_name` (the client sanitizes file names as well) and/or throw an error. I like that returning an empty string makes it very easy to pick between `file_name` and `sanitized_file_name` using `or`, but maybe throwing and error signals the intent more clearly.

Reply:
What if the filename is `CON`? Why do we bother to sanitize it if we're then going to allow that filename (by doing `file_name or sanitized_file_name` as you suggest)?

I'm leaning more towards throwing an error. @jackrobison @tzarebczan what do you guys think?

Reply:
Because maybe we should at least try to save the file using the original file name before failing, at least when downloading?

Reply:
I don't think a weird name should cause the download to fail. If the name gets sanitized down to nothing, it should probably default to `_`, or something more descriptive like `lbry_download`